pkg/query/endpointset.go

```diff
@@ -344,6 +360,15 @@ func NewEndpointSet(
 		},
 		endpoints: make(map[string]*endpointRef),
 	}
+
+	// Log the endpoints
+	{
+		level.Info(es.logger).Log("msg", "All gPRC endpoints for querying")
+		for _, s := range endpointSpecs() {
+			level.Info(es.logger).Log("msg", "One gPRC endpoint for querying", "address", s.addr, "strict", s.isStrictStatic)
```

Comment:
make it debug only to avoid it is too noisy, @mingzhao-db complained about our log volume previously

Reply:
This function is called once to create a new endpoint set.
```
[dev-aws-eu-west-1] [pantheon] [pantheon-range-querier-deployment-57f975f46c-8l7gn] > logs | rg 'All gPRC endpoints for querying'
ts=2024-04-05T00:45:36.699595553Z caller=endpointset.go:366 level=info name=pantheon-range-querier component=endpointset msg="All gPRC endpoints for querying"
```